tests/test_oic.py

```diff
@@ -632,8 +632,20 @@ def test_construct_EndSessionRequest(self):
 
         self.client.grant["foo"].tokens.append(Token(resp))
 
+        # state only in kwargs
         args = {"redirect_url": "http://example.com/end"}
         esr = self.client.construct_EndSessionRequest(state="foo", request_args=args)
+        assert _eq(esr.keys(), ["id_token", "redirect_url"])
```

Comment:
Please make these as a separate tests under the same class. It is then much cleaner to figure out what fails.

Reply:
OK

Reply:
Also shouldn't this include `state` as well?

Reply:
No, `state` in kwargs are different from `state` in request_args.
This comes from the fact that we have state in different requests being different things.
`state` in an authorisation request is NOT the same thing as `state` in an end session requests.
The `state` that comes in the authorisation request are used internally as a key to information resources so the construct_EndSessionRequest needs that `state`. The `state` in the request_args is specific to the end session request and is not expected to have the same value as the authorisation request state.

Reply:
Hence, `state` in kwargs should not bleed over into the request args.

Reply:
OK.